libfwupdplugin/fu-device.c

```diff
@@ -1859,6 +1859,21 @@ fu_device_get_physical_id (FuDevice *self)
 void
 fu_device_add_flag (FuDevice *self, FwupdDeviceFlags flag)
 {
+	/* none is not used as an "exported" flag */
+	if (flag == FWUPD_DEVICE_FLAG_NONE)
+		return;
+
+	/* being both a bootloader and requiring a bootloader is invalid */
+	if (flag == FWUPD_DEVICE_FLAG_NONE ||
```

Comment:
aren't you not going to hit this clause since you checked for none and returned earlier?

Reply:
Urgh, I missed the bit where I remove the ugly NONE hack.

Reply:
Okay, respun, thanks.